ui/src/components/connections/ConnectionDetails.tsx

```diff
@@ -60,16 +164,64 @@ export function ConnectionDetails() {
     [connectionID, env]
   );
 
+  const fetchCredentials = useCallback(
+    async (signal: AbortSignal) => {
+      if (isAsyncTaskDataAvailable(connection)) {
+        setCredentials({ status: "loading" });
+        const { userID } = connection.data;
+        const response = await getCredentials({
+          env,
+          params: {
+            // TODO should change when PID-498 is done
+            query: query ? `${userID} ${query}` : `${userID}`,
+            type: credentialType,
+          },
+          signal,
+        });
+        if (response.isSuccessful) {
+          setCredentials({
+            data: response.data,
+            status: "successful",
+          });
+        } else {
+          if (!isAbortedError(response.error)) {
+            setCredentials({ error: response.error, status: "failed" });
+          }
+        }
+      }
+    },
+    [connection, env, query, credentialType]
+  );
+
+  const handleTypeChange = ({ target: { value } }: RadioChangeEvent) => {
+    const parsedTypeValue = credentialTypeParser.safeParse(value);
+    if (parsedTypeValue.success) {
+      setCredentialType(parsedTypeValue.data);
+    } else {
+      processZodError(parsedTypeValue.error).forEach((error) => void message.error(error));
+    }
+  };
+
   useEffect(() => {
     const { aborter } = makeRequestAbortable(fetchConnection);
 
     return aborter;
   }, [fetchConnection]);
 
-  // TODO PID-481
-  const credentialsList = []; /* isAsyncTaskDataAvailable(connection) && connection.data.credentials
-      ? connection.data.credentials
-      : []; */
+  useEffect(() => {
+    const { aborter } = makeRequestAbortable(fetchCredentials);
+
+    return aborter;
+  }, [fetchCredentials]);
+
+  const credentialsList = isAsyncTaskDataAvailable(credentials)
+    ? credentials.data
+    : isAsyncTaskDataAvailable(connection) && !hasAsyncTaskFailed(credentials)
+    ? connection.data.credentials
+    : [];
```

Comment:
I'm a bit lost here...

  * Why is this condition required? `&& !hasAsyncTaskFailed(credentials)`
  * Why are we getting the credentials from 2 different places?

Reply:
I'd forget about the credentials in the connection and fetch them from their endpoint, so we can filter them and there's a single source of truth.